### PR TITLE
Update wc-product-functions.php

### LINKED ISSUE #27012: is_vat_exempt removes VAT from shop pages only if prices are entered inclusive of tax
Please provide us with the information requested in this bug report. Without these details, we won't be able to fully evaluate this issue. 
Bug reports lacking detail, or for any other reason than to report a bug, may be closed without action.

**Prerequisites (mark completed items with an [x]):**
- [x] I have have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

**Describe the bug**
I have programatically set a user to be tax exempt using the woocommerce set_is_vat_exempt() function.
This correctly removes VAT from the Cart pages. However, in the shop pages, it removes VAT only if the shop setting is "I will enter prices inclusive of tax".

**Expected behavior**
The way a user enters prices (inclusive or exclusive of tax), should have no relation whatsoever to how tax is displayed. I would expect WooCommerce to remove tax in shop pages for VAT exempt users regardless of how the shop owner enters prices.

**Actual behavior**
WooCommerce only removes tax in shop pages if the setting is "I will enter prices inclusive of tax".

**Steps to reproduce the bug (We need to be able to reproduce the bug in order to fix it.)**
Steps to reproduce the bug:
1. Set current user as vat exempt programatically: WC()->customer->set_is_vat_exempt(true);
2. Set the following settings: 
A) Display prices in the shop including tax. 
B) Display prices in the cart including tax. 
C) I will enter prices inclusive of tax

3. Create a product with the price of 100 and a tax rate of 15% for testing purposes.
3.5. Edited: Also please set "Calculate tax based on Shop Settings" (and set the 15% tax rate for the country the shop is based in)
4. Price appears correctly with the tax removed: 86.96
5. Change I will enter prices inclusive of tax to " I will enter prices exclusive of tax"
6. The price that should now appear in shop pages is 100. However it is 115. The tax is no longer being removed by the is_vat_exempt status

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
We use the [WooCommerce System Status Report](https://docs.woocommerce.com/document/understanding-the-woocommerce-system-status-report/) to help us evaluate the issue. 
Without this report we won't be able to fully evaluate this issue.
<details>
`
### WordPress Environment ###

WordPress address (URL): http://localhost:10093
Site address (URL): http://localhost:10093
WC Version: 4.3.0
REST API Version: ✔ 1.0.10
WC Blocks Version: ✔ 2.7.1
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ✔ 1.3.0
Log Directory Writable: ✔
WP Version: 5.4.2
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.16.0
PHP Version: 7.3.5
PHP Post Max Size: 1,000 MB
PHP Time Limit: 1200
PHP Max Input Vars: 4000
cURL Version: 7.64.0
OpenSSL/1.1.1b

SUHOSIN Installed: –
MySQL Version: 8.0.16
Max Upload Size: 300 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 4.3.0
WC Database Prefix: wp_
Total Database Size: 3.74MB
Database Data Size: 2.34MB
Database Index Size: 1.40MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 1.50MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_posts: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 1
b2bking_custom_field: 9
b2bking_custom_role: 2
b2bking_rule: 1
page: 6
post: 2
product: 2

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (1) ###

WooCommerce: by Automattic – 4.3.0

### Inactive Plugins (1) ###

B2BKing: by WebWizards – 1.8.0

### Settings ###

API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: left
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #19 - /shop/
Cart: #20 - /cart/
Checkout: #21 - /checkout/
My account: #22 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 2.5.8
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 5
Oldest: 2020-07-10 12:49:11 +0000
Newest: 2020-07-10 13:39:16 +0000

Pending: 1
Oldest: 2020-07-11 12:49:11 +0000
Newest: 2020-07-11 12:49:11 +0000

`
</details>

**Solving this**
I suppose you could go about solving this in 2 ways, but I would ask that you please address this in a way that offers an option to consistently remove tax from shop pages (perhaps add a parameter to the set_is_vat_exempt function regarding display of VAT in shop pages).

This option is extremely important for legal reasons for business-to-business e-commerce, particularly in Europe.

### PR DESCRIPTION
**Pull request by kingplugins**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #27012  .

How prices are entered (including or excluding tax) should have no relation to how prices are displayed.
In the current code, a user's vat exempt status only removes tax in shop pages if prices are entered including tax
This proposed change also removes tax if prices are entered excluding tax, thus making this behaviour consistent

### How to test the changes in this Pull Request:

1. Set current user as vat exempt programatically: WC()->customer->set_is_vat_exempt(true);
2. Set the following settings:
   A) Display prices in the shop including tax.
   B) Display prices in the cart including tax.
   C) I will enter prices exclusive of tax
3. Create a product with the price of 100 and a tax rate of 15% for testing purposes.
4. The price that will now appear in shop pages is 100. Previously it was 115, and the code did not account for the user's tax exempt status in shop pages.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> User's VAT Exempt status now removes VAT from shop pages when prices are entered excluding tax

### CODE REVIEW

File: includes/wc-product-functions.php [hunk 1]
```diff
@@ -982,7 +982,14 @@ function wc_get_price_including_tax( $product, $args = array() ) {
 				$taxes_total = array_sum( array_map( 'wc_round_tax_total', $taxes ) );
 			}
 
-			$return_price = round( $line_price + $taxes_total, wc_get_price_decimals() );
+			/**
+			 * If the customer is excempt from VAT, remove the taxes here.
+			*/
+			if ( ! empty( WC()->customer ) && WC()->customer->get_is_vat_exempt() ) { //
```
Comment (rrennick) [inline includes/wc-product-functions.php:988]: @stefanstatescu1 Thanks for submitting this PR. This check would work better if it were the condition before `if ( 'yes' === get_option( 'woocommerce_tax_round_at_subtotal' ) ) {` and set `$taxes_total = 0.00`. That way we would avoid calculating the tax total when it wasn't used. The PR also has coding standards issues.

**Review by rrennick — CHANGES_REQUESTED** [review]

File: includes/wc-product-functions.php (hunk 1 above)
Comment (stefanstatescu1) [reply includes/wc-product-functions.php]: Thanks @rrennick 
I think it's better now?

**Comment by rrennick** [comment]
@stefanstatescu1 I apologize. I should have tested `master` to see if I could reproduce the issue. In my install I have the correct prices in

Single product
Screen Shot 2020-07-14 at 12 45 15 PM

Category archive
Screen Shot 2020-07-14 at 12 45 30 PM

Cart
Screen Shot 2020-07-14 at 12 40 46 PM

Cart guest customer
Screen Shot 2020-07-14 at 12 43 11 PM

**Comment by stefanstatescu1** [comment]
@rrennick I've done a lot of tests over the past 2 hours, all tests with fresh WP 5.4.2 installs, fresh WooCommerce installs (from wp repo), Storefront theme, no other plugins. I can reproduce this consistently but I also got your result at some point.

Long story short, can you please try the same test you tried above, but set:
**Calculate tax based on "Shop address". (and set a tax rate of 15% for the country you set the shop to be based in).**

I'm pretty sure that in the test you did above you got that result not because of the user's tax exemption, but because WooCommerce didn't consider the user to be in a billing/shipping area for which tax applied (but it did for the guest user).

**Comment by rrennick** [comment]
@stefanstatescu1 I changed both customer addresses to be in the tax zone. I get the same result with any choice on where to base tax calculations on. Have you confirmed that the tax exempt setting is being saved to the user meta table?

**Comment by stefanstatescu1** [comment]
I'm not sure where this could be seen in the user meta table. I had checked the get_is_vat_exempt() function and it returns 1.

As I switch between "enter prices inclusive of tax" and "enter prices exclusive of tax", the price for the item switches between 8.70 (tax exemption working) and 11.50 (tax exemption not working)  

Can you please check that if you change to prices entered inclusive of tax, you get 8.70? my guess is that you might get 10.

If all works for you, I'll go through this again tomorrow and re-check everything.

**Comment by stefanstatescu1** [comment]
I have tested this again and documented the exact steps I am taking in order. Thanks for your time.

<details>

1) Fresh WP 5.4.2 install via Local by Flywheel
2) WooCommerce install and activate from the WP Repo (Plugins->Add New) 
3) Quick WooCommerce setup wizard + activate Storefront
4) WooCommerce settings: Enable Tax,  I will enter prices exclusive of tax, Display prices everywhere including tax, set tax rate for the country of the store 15%, calculate tax based on shop address: https://prnt.sc/ti36l7
5) Go to functions.php add:
add_action('init', function(){
	if (isset(WC()->customer)){
		WC()->customer->set_is_vat_exempt(true);
	}
});
6) Create a simple product with regular price 10
7) Go to Shop page-> price is 11.50: https://prnt.sc/ti2sie
8) Change "enter prices exclusive of tax" to "enter prices inclusive of tax"
9) Refresh Shop page-> price is 8.70: https://prnt.sc/ti2t23
-----------------
More checks:
10) I check with a different browser, same results
11) I change "true" to "false" in the code in functions.php to remove the exemption for the current session. 
12) Prices are now 10 for "prices entered inclusive of tax" and 11.50 for "prices entered exclusive of tax".
13) Change "false" to "true" again. Prices now again switch between 8.70 and 11.50
14) I comment out the code in functions.php (but the tax exemption is still applied for the session). I add the following code to functions.php:

add_action('wp_footer', function(){
	if (isset(WC()->customer)){
		update_option('debug_test', WC()->customer->get_is_vat_exempt());
	}
});

See: https://i.imgur.com/yKP2Ll8.png

15) I refresh the shop page. Price is still 11.50
wp_options shows 'debug_test' to be 1:
https://prnt.sc/ti31h3

16) I change "exclusive of tax" to "inclusive of tax"
Pirce is 8.70
'debug_test' is 1
17) I use set_is_vat_exempt(false), and debug_test becomes null / empty: https://prnt.sc/ti33dn
18) I also add the addresses of the user in my account, it does not affect results in any way
</details>

**Comment by rrennick** [comment]
> Can you please check that if you change to prices entered inclusive of tax, you get 8.70? my guess is that you might get 10.

I get $8.70 for the exempt user and $10 for guest.

> I'm not sure where this could be seen in the user meta table

DB table `wp_usermeta`, `meta_key` is `is_vat_exempt`. If the row is there for your test user it should contain either `yes` or `no`.

**Comment by stefanstatescu1** [comment]
> DB table `wp_usermeta`, `meta_key` is `is_vat_exempt`. If the row is there for your test user it should contain either `yes` or `no`.

I don't have such a meta_key in the wp_usermeta table. I don't believe this is correct, as the vat exempt status is per session, not permanent user data.
Clearing sessions either in WooCommerce->Status->Tools or deleting the current session in wp_woocommerce_sessions clears the vat exempt status.

**Comment by rrennick** [comment]
> I don't have such a meta_key in the wp_usermeta table.

@stefanstatescu1 Could you test adding it for your test user with a value of `yes`?

**Comment by stefanstatescu1** [comment]
> > I don't have such a meta_key in the wp_usermeta table.
> 
> @stefanstatescu1 Could you test adding it for your test user with a value of `yes`?

@rrennick I had tested that, it seemed to have no effect /interaction with anything. I also gave it random values like "123" to see if set_is_vat_exempt(true or false) would change the value, but it did not.

**Comment by rrennick** [comment]
Which hook did you use for `WC()->customer->set_is_vat_exempt(true);` ?

**Comment by stefanstatescu1** [comment]
I used init
```
add_action('init', function(){
   if (isset(WC()->customer)){
      WC()->customer->set_is_vat_exempt(true);
   }
});
```

**Review by rrennick — APPROVED** [review]
@stefanstatescu1 Thanks for the fix. I apologize for the long delay in getting back to this PR.

**Comment by peterfabian** [comment]
Can we please get this another quick look before merging? I'm not sure why it wasn't merged, but it seems to me it was ready to go back then?

**Comment by peterfabian** [comment]
Hi @stefanstatescu1 ! Sorry for the large delay in response. It would be really nice if this was covered by a unit test or two. Would you be willing to add these and rebase the PR, please?

**Comment by jeffpaul** [comment]
@iamdharmesh given this relates to the EU VAT item you worked on, mind checking into the unit test(s) that ideally get added here?

**Comment by iamdharmesh** [comment]
@jeffpaul I have created a new PR for that.

**Comment by samueljseay** [comment]
From what I understand this was superceded by https://github.com/woocommerce/woocommerce/pull/33991 so I will close it .

If you feel this has been done by mistake then please re-open with description of the reasoning 😃 


Thanks!

**Pull request closed without merging**